proto/batch_plan.proto

```diff
@@ -109,6 +109,13 @@ message HashAggNode {
   repeated expr.AggCall agg_calls = 2;
 }
 
+message ExpandNode {
+  message Keys {
+    repeated uint32 keys = 1;
+  }
+  repeated Keys expanded_keys = 1;
+}
```

Comment:
Do you mean `Keys`?

Reply:
Yes, IIUC it's just `InputRef`, right?

Reply:
It's index of column actually......

Reply:
They are quite similar to me 😂

Reply:
Yes `InputRef` is essentially the same as `uint32` here. But I do not see why we need to generalize from `InputRef` to arbitrary `Expr`, or how this is related to `HopWindowExecutor `.

As for `uint32` vs `InputRef`, I prefer to avoid `InputRef`: it is lacking the `return_type` part to be treated as a genuine `Expr`, and the caller (ExecutorBuilders including hop) actually just takes `column_idx` out of it rather than building it into an expression and then evaluate.

Reply:
The expand operator are following expressions:
```
[
  [InputRef(1), InputRef(2)],
  [InputRef(3), InputRef(4)]
]
```

If we change HopWindowExecutor to ExpandExecutor, they are just following expressions:
```
[
  [window start expr in interval 1, window end expr in interval 1, .... original exprs],
  [window start expr in interval 2, window end expr in interval 2, .... original exprs],
 [window start expr in interval 3, window end expr in interval 3, .... original exprs],
           .
           .
]
```

And these can be computed by frontend, which can be reused by both streaming and batch executors. As with data type, I don't see it as a blocking issue for us.

Reply:
I’ve got it and +1 for this. maybe we can have a try after this PR is merged and well tested with distinct agg.

Reply:
This is an interesting generalization. We can give it a try, but probably after we have a working e2e distinct agg as tianshuo suggested above.

Actually, this generalized node is just a multi-row expr evaluation. If we want to, it can also cover use cases of `project` (single row of exprs) and `values` (input is fixed to 1-row-0-col). Just not sure if this generalization is too much.

Also to my understanding, it would be
```
[
  [LiteralNull, InputRef(1), InputRef(2), LiteralNull, LiteralNull],
  [LiteralNull, LiteralNull, LiteralNull, InputRef(3), InputRef(4)]
]
```
rather than the example above.

I also agree that data type is not a big issue. I was just listing the differences between our InputRef in rust vs InputRef in proto.

Reply:
I'm ok to try it after distinct agg has been resolved.